### PR TITLE
(+semver: feature) EN-4822: Add several features

### PR DESCRIPTION
**Pull request by thzinc**
This is based on @annymsMthd's initial PR #1. I can rebase as needed.

Several updates;

* Added documentation (Common DX, Cobra command auto-generated docs, etc.)
* Updated configuration
  * Added ignore support (uses `.idlignore` by default, may be overridden in `idl.yaml`)
  * Added dependency-level repository so one configuration can represent dependencies in multiple idl-repositories
* Updated idl-repository service
  * Handled OS signals properly
  * Added logging with `logrus`
  * Updated Dockerfile to expose its port and volume

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,87 @@
...
+[idl-pull]: docs/idl/idl_pull.md
+[idl-push]: docs/idl/idl_push.md
+[idl-repository]: docs/idl-repository/idl-repository.md
+
+## Building
+
+[![Travis](https://img.shields.io/travis/syncromatics/idl-repository.svg)](https://travis-ci.org/syncromatics/idl-repository)
+[![Docker Build Status](https://img.shields.io/docker/build/syncromatics/idl-repository.svg)](https://hub.docker.com/r/syncromatics/idl-repository/)
+
+Ensure you have your `GOPATH` configured properly. (Typically, you'll want to check this repo out to `$(go env GOPATH)/src/github.com/syncromatics/idl-repository`.) You'll also need Docker to build the Docker image for the repository.
```
Comment (annymsMthd) [inline README.md]: Since this is using go modules having it in the go path is no longer a requirement.

Comment (thzinc) [reply README.md]: Done

File: makefile [hunk 2]
```diff
@@ -5,13 +5,24 @@ WINDOWS_BUILD_PATH = $(BUILD_PATH)/windows/idl.exe
 MAC_BUILD_PATH = $(BUILD_PATH)/darwin/idl
 
 export VERSION=$(shell gogitver)
+export COMMIT_HASH=$(shell git rev-parse HEAD)
+export BUILD_DATE=$(shell date +%Y-%m-%dT%T%z)
+export BUILD_FLAGS=\
+	-X github.com/syncromatics/idl-repository/cmd/idl/cmd.version=$(VERSION) \
```
Comment (annymsMthd) [inline makefile]: I gotta remember this one. Awesome way to get the version in there.

**Comment by annymsMthd** [comment]
Not sure what is up with the push build. Looks like gogitver is the issue

File: makefile (hunk 2 above)
Comment (thzinc) [reply makefile]: go-version turned me on to build variables: https://github.com/christopherhein/go-version

It's pretty cool, but it took me a while to find the docs that led to using the full package/import path to make it work.

**Comment by thzinc** [comment]
> Not sure what is up with the push build. Looks like gogitver is the issue

Travis defaults to pulling a depth of 50 commits, and doesn't usually include the master branch. I had already fixed this in our DX Travis template for .NET Core. Seems like a Go template might be useful here.

**Pull request merged**